WordPress/src/main/java/org/wordpress/android/viewmodel/PluginBrowserViewModel.kt

```diff
@@ -0,0 +1,465 @@
+
+    private fun updateAllPluginListsIfNecessary() {
+        val copiedSet = HashSet(updatedPluginSlugSet)
+        handler.postDelayed({
+            // Using the size of the set for comparison might fail since we clear the updatedPluginSlugSet
+            if (copiedSet == updatedPluginSlugSet) {
+                updateAllPluginListsWithNewPlugins(copiedSet)
+                updatedPluginSlugSet.clear()
+            }
+        }, 250)
+    }
+
+    private fun updateAllPluginListsWithNewPlugins(updatedPluginSlugSet: Set<String>) {
+        if (updatedPluginSlugSet.isEmpty()) {
+            return
+        }
+        val newPluginMap = HashMap<String, ImmutablePluginModel>(updatedPluginSlugSet.size)
+        for (slug in updatedPluginSlugSet) {
+            val immutablePlugin = mPluginStore.getImmutablePluginBySlug(site!!, slug)
+            if (immutablePlugin != null) {
+                newPluginMap[slug] = immutablePlugin
+            }
+        }
+        // By combining all the updated plugins into one map, we can post a single update to the UI after changes are
+        // reflected
+        updatePluginListWithNewPlugin(_featuredPlugins, newPluginMap)
+        updatePluginListWithNewPlugin(_newPlugins, newPluginMap)
+        updatePluginListWithNewPlugin(_popularPlugins, newPluginMap)
+        updatePluginListWithNewPlugin(_searchResults, newPluginMap)
+
+        // Unfortunately we can't use the same method to update the site plugins because removing/installing plugins can
+        // mess up the list. Also we care most about the Site Plugins and using the store to get the correct plugin
+        // information is much more reliable than any manual update we can make
+        reloadPluginDirectory(PluginDirectoryType.SITE)
+    }
+
+    private fun updatePluginListWithNewPlugin(mutableLiveData: MutableLiveData<List<ImmutablePluginModel>>,
+                                              newPluginMap: Map<String, ImmutablePluginModel>) {
+        val pluginList = mutableLiveData.value
+        if (pluginList == null || pluginList.isEmpty() || newPluginMap.isEmpty()) {
+            // Nothing to update
+            return
+        }
+        // When a site or wporg plugin is updated we need to update every occurrence of that item
+        val newList = ArrayList<ImmutablePluginModel>(pluginList.size)
+        var isChanged = false
+        for (immutablePlugin in pluginList) {
+            val slug = immutablePlugin.slug
+            val newPlugin = newPluginMap[slug]
+            if (newPlugin != null) {
+                // add new item
+                newList.add(newPlugin)
+                isChanged = true
+            } else {
+                // add old item
+                newList.add(immutablePlugin)
+            }
+        }
+        // Only update if the list is actually changed
+        if (isChanged) {
+            mutableLiveData.postValue(newList)
+        }
+    }
+
+    private fun shouldSearch(): Boolean {
```

Comment:
You could inline simple functions like this: `private fun shouldSearch() = searchQuery.length > 1`. But since there is no parameter, it makes more sense to make it a private property:
```
private val shouldSearch: Boolean
        get() = searchQuery.length > 1
```

Reply:
Fixed in fdb98adf2ba9bf52bd7d6b71f4f9f04abe6e4703.